dataland-backend/src/main/kotlin/org/dataland/datalandbackend/model/eutaxonomy/financials/EuTaxonomyDataForFinancials.kt

```diff
@@ -19,7 +19,8 @@ import java.util.EnumSet
  * --- API model ---
  * Fields of the questionnaire for the EuTaxonomyForFinancials framework
  */
-@DataType("eutaxonomy-financials")
+@Suppress("MagicNumber")
+@DataType("eutaxonomy-financials", 1)
```

Comment:
Das mit der MagicNumber ist etwas unschön. Es hat aus meiner Sicht auch den Nachteil das man nicht auf einen Blick irgendwo sieht was die Reihenfolge eigentlich ist. Wie wäre ist mit folgender Idee: Wir definieren eine Liste die einfach aus den Namen der DataTypes besteht in der Reihenfolge wie gewünscht. An dieser Stelle hier muss dann list.indexOf("WasAuchImmerDerNameIst")+1 stehen statt der magic number. Dann wäre man das suppress los und man sieht an einer znetralen Stelle was die Reihenfolge sein soll. Es kann dann auch nicht vorkommen das Zahlen doppelt vergeben werden. Man muss die Liste aber manuell pflegen.

Reply:
Ich habe mich für diese Lösung entschieden, um bei dem Design: "Alles was mit dem Framework zu tuen hat wird möglichst in der Framework-Toolbox definiert" zu bleiben. Weiterhin ist ein Annotation-Basiertes Ordering bei Depdency-Injection frameworks üblich (also zumindest in Spring). Ich finde da zentralisierte Listen blöd, weil man die garantiert vergisst. Die resultierende Reihenfolge sieht man unkompliziert im OpenAPI spec oder im Frontend.